target/snitch_cluster/sw/apps/snax-streamer-gemm/src/snax-streamer-gemm.c

```diff
@@ -36,13 +34,13 @@ int main() {
     // Wait for DMA to finish
     snrt_cluster_hw_barrier();
 
-    if (snrt_is_compute_core()) {
-        uint32_t gemm_start = snrt_mcycle();
+    if (snrt_global_core_idx() == 0) {
+        // uint32_t gemm_start = snrt_mcycle();
 
         // Set Streamer configuration CSR
-        set_streamer_csr(K, N, M, strideInnermostA, ldA, strideInnermostB, ldB,
-                         strideInnermostC, ldC, delta_local_a, delta_local_b,
-                         delta_local_c);
+        set_streamer_csr(K, N, M, strideInnermostA, ldA, spatialA,
+                         strideInnermostB, ldB, spatialB, strideInnermostC, ldC,
+                         spatialC, delta_local_a, delta_local_b, delta_local_c);
         // Set CSR to start Streamer
         set_streamer_start();
```

Comment:
Dumb question: We can already setup the streamer, while the block gemm is not yet consuming the outputs of the streamer?
I would think it makes more sense to set all CSR settings first, and only then launch everything. But I'm assuming this is more optimal?

Reply:
>Dumb question: We can already setup the streamer, while the block gemm is not yet consuming the outputs of the streamer?

Yes.

>I would think it makes more sense to set all CSR settings first, and only then launch everything. But I'm assuming this is more optimal?

We can start the streamer earlier so the streamer can already do some prefetch.

Reply:
Cool thanks!